### PR TITLE
audits(is-crawlable): line number for blocking directive in robots.txt

### LINKED ISSUE #4877: Surface which line in robots.txt is blocking indexing
CLI said the page is blocked for indexing but CDT said it isn't blocked, please see the video http://nimb.ws/oXGYa3.

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #4877

### CODE REVIEW

File: lighthouse-core/audits/seo/is-crawlable.js [hunk 1]
```diff
@@ -122,10 +122,14 @@ class IsCrawlable extends Audit {
...
           if (!robotsTxt.isAllowed(mainResource.url)) {
+            const line = robotsTxt.getMatchingLineNumber(mainResource.url);
             blockingDirectives.push({
               source: {
-                type: /** @type {'url'} */ ('url'),
-                value: robotsFileUrl.href,
+                type: /** @type {'source-location'} */ ('source-location'),
+                url: robotsFileUrl.href,
+                urlProvider: /** @type {'network'} */ ('network'),
+                line: line !== undefined ? line - 1 : 0,
```
Comment (patrickhulce) [inline lighthouse-core/audits/seo/is-crawlable.js]:
```suggestion
                line: typeof line === 'number' ? line - 1 : 0,
```

super nit, just a tad clearer IMO

File: lighthouse-core/audits/seo/is-crawlable.js [hunk 2]
```diff
@@ -122,10 +122,14 @@ class IsCrawlable extends Audit {
           const robotsTxt = robotsParser(robotsFileUrl.href, artifacts.RobotsTxt.content);
 
           if (!robotsTxt.isAllowed(mainResource.url)) {
+            const line = robotsTxt.getMatchingLineNumber(mainResource.url);
             blockingDirectives.push({
               source: {
-                type: /** @type {'url'} */ ('url'),
-                value: robotsFileUrl.href,
+                type: /** @type {'source-location'} */ ('source-location'),
```
Comment (patrickhulce) [inline lighthouse-core/audits/seo/is-crawlable.js]: this falls back to URL in non-devtools, right?

Comment (connorjclark) [reply lighthouse-core/audits/seo/is-crawlable.js]: correct (see https://github.com/GoogleChrome/lighthouse/pull/9354 for images)

File: lighthouse-core/audits/seo/is-crawlable.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/audits/seo/is-crawlable.js]: wb `const line = robotsTxt.getMatchingLineNumber(mainResource.url) || 1;` and `line: line - 1`

Comment (patrickhulce) [reply lighthouse-core/audits/seo/is-crawlable.js]: wfm 👍

**Pull request merged**